src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -137,17 +146,17 @@ internal class AuthorityInfo
         internal bool IsTenantOverrideSupported => AuthorityType == AuthorityType.Aad;
         internal bool IsMultiTenantSupported => AuthorityType != AuthorityType.Adfs;
         internal bool IsClientInfoSupported => AuthorityType == AuthorityType.Aad || AuthorityType == AuthorityType.Dsts || AuthorityType == AuthorityType.B2C;
+        internal string DiscoveryDocumentAddress { get; set; }
```

Comment:
AuthorityInfo is a common object for all authorities. Would `DiscoveryDocumentAddress` be better placed in GenericAuthority class?

Reply:
@bgavrilMS it looks like the Authority instance which is set into Config when using the builder is not the same one as the one used when requesting a token causing GenericAuthority properties to be null when requesting a token. Is it the normal behavior ? Should I try to stabilize the Config.Authority instance between builder usage and fetching the token ?